src/couch_replicator/test/couch_replicator_create_target_with_options_tests.erl

```diff
@@ -0,0 +1,168 @@
+            [{P, fun should_create_target_with_q_8/2} || P <- Ps] ++
+            [{P, fun should_create_target_with_q_default/2} || P <- Ps] ++
+            [{P, fun should_not_create_target_with_q_any/2} || P <- Ps]
+        }
+    }.
+
+
+should_create_target_with_q_1({From, To}, {_Ctx1, _Ctx2, {Source, Target}}) ->
+    RepObject = {[
+        {<<"source">>, db_url(From, Source)},
+        {<<"target">>, db_url(To, Target)},
+        {<<"create_target">>, true},
+        {<<"create_target_params">>, {[{<<"q">>, <<"1">>}]}}
+    ]},
+    create_db(From, Source),
+    create_doc(From, Source),
+    {ok, _} = couch_replicator:replicate(RepObject, ?ADMIN_USER),
+
+    {ok, TargetInfo} = fabric:get_db_info(Target),
+    {ClusterInfo} = couch_util:get_value(cluster, TargetInfo),
+    delete_db(From, Source),
+    delete_db(To, Target),
+    ?_assertEqual(1, couch_util:get_value(q, ClusterInfo)).
+
+
+should_create_target_with_q_4({From, To}, {_Ctx1, _Ctx2, {Source, Target}}) ->
+    RepObject = {[
+        {<<"source">>, db_url(From, Source)},
+        {<<"target">>, db_url(To, Target)},
+        {<<"create_target">>, true},
+        {<<"create_target_params">>, {[{<<"q">>, <<"4">>}]}}
+    ]},
+    create_db(From, Source),
+    create_doc(From, Source),
+    {ok, _} = couch_replicator:replicate(RepObject, ?ADMIN_USER),
+
+    {ok, TargetInfo} = fabric:get_db_info(Target),
+    {ClusterInfo} = couch_util:get_value(cluster, TargetInfo),
+    delete_db(From, Source),
+    delete_db(To, Target),
+    ?_assertEqual(4, couch_util:get_value(q, ClusterInfo)).
+
+
+should_create_target_with_q_8({From, To}, {_Ctx1, _Ctx2, {Source, Target}}) ->
+    RepObject = {[
+        {<<"source">>, db_url(From, Source)},
+        {<<"target">>, db_url(To, Target)},
+        {<<"create_target">>, true},
+        {<<"create_target_params">>, {[{<<"q">>, <<"8">>}]}}
+    ]},
+    create_db(From, Source),
+    create_doc(From, Source),
+    {ok, _} = couch_replicator:replicate(RepObject, ?ADMIN_USER),
+
+    {ok, TargetInfo} = fabric:get_db_info(Target),
+    {ClusterInfo} = couch_util:get_value(cluster, TargetInfo),
+    delete_db(From, Source),
+    delete_db(To, Target),
+    ?_assertEqual(8, couch_util:get_value(q, ClusterInfo)).
+
+
+should_create_target_with_q_default(
+    {From, To}, {_Ctx1, _Ctx2, {Source, Target}}) ->
+    RepObject = {[
+        {<<"source">>, db_url(From, Source)},
+        {<<"target">>, db_url(To, Target)},
+        {<<"create_target">>, true}
+    ]},
+    create_db(From, Source),
+    create_doc(From, Source),
+    {ok, _} = couch_replicator:replicate(RepObject, ?ADMIN_USER),
+
+    {ok, TargetInfo} = fabric:get_db_info(Target),
+    {ClusterInfo} = couch_util:get_value(cluster, TargetInfo),
+    Q = config:get("cluster", "q", "8"),
+    delete_db(From, Source),
+    delete_db(To, Target),
+    ?_assertEqual(list_to_integer(Q), couch_util:get_value(q, ClusterInfo)).
+
+
+should_not_create_target_with_q_any(
```

Comment:
This test case is used to prevent from regression. If `create_target` is not specified or set to false, there should not be target database to be created, even if there is any value set for `create_target_params`.